src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -0,0 +1,191 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.ClassificationCategory;
+import com.google.cloud.language.v1.ClassifyTextRequest;
+import com.google.cloud.language.v1.ClassifyTextResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.language.v1.EntityMention;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.LanguageServiceSettings;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/*
+* This class creates objects that analyse text by 
+* extracting the categories, the mood, the events.
+* The scope is to get a final set of the key words
+* from the text. 
+*/
+
+public final class TextAnalyser {
+  private final String message;
+  private static final double EPSILON = 0.00001;
+
+  public TextAnalyser(String message) {
+     this.message = message.toLowerCase();
+  }
+
+  public final LanguageServiceSettings getSettings() throws IOException {
+    HeaderProvider headerProvider =
+            FixedHeaderProvider.create("X-Goog-User-Project","google.com:gpostcard");
+    return LanguageServiceSettings.newBuilder()
+                .setHeaderProvider(headerProvider)
+                .build();
+  }
+
+  public float getSentimentScore() throws IOException {
+    try (LanguageServiceClient languageService = LanguageServiceClient.create(getSettings())) {
+      Document doc =
+        Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+        
+      return sentiment.getScore();
+    }
+  }
+
+  public String getMood() throws IOException {
+    float score = getSentimentScore();
+
+    if (Math.abs(score - 1) < EPSILON) {
+      return "very happy";
+    }
+ 
+    if (Math.abs(score + 1) < EPSILON) {
+      return "so pessimistic";
+    }
+ 
+    int position = 0;
+    String[] moods = new String[]{"neutral", "calm", "relaxed", "serene", "contented",
+                                  "joyful", "happy", "delighted", "excited", "thrilled",
+                                  "tense", "nervous", "stressed", "upset", "sad", 
+                                  "depressed", "bored", "fatigued", "pessimisctic"};
+ 
+    score = score * 10;
```

Comment:
on quick read, it's not super-clear to me how you get to the "magic" numbers **9** and **10** here. mood contains **19** entries, surely no accident. An assertion in the code (https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html) could help documenting the relationship, e.g. what if I added an entry later to "mood" array?

Reply:
The length of moods array should not drop 20; if it increases, the accuracy of the mood selection won't be affected. 
If I say something like `assert moods.length == 20 : "There can only be 20 moods.";` is it okay? @jeadorf 
Also, here is a photo of the representation of the moods on the x axis and how I got the 9th and 10th position in the array. Maybe that can give a better explanation.
![github](https://user-images.githubusercontent.com/37437735/91044514-142b2280-e61e-11ea-94fb-933b8856cc60.jpg)

Reply:
sure ... nice drawing!